lib/tasks/connectivity.rake

```diff
@@ -0,0 +1,172 @@
+
+# Checks backend connectivity to the various VA machines. Run with
+# `RAILS_ENV=production bundle exec rake connectivity:all`
+# Also does sanity check to ensure that logs directory is writeable
+
+# Allows running on development machines
+Rails.logger = Logger.new(STDOUT)
+
+# For Rx/SM
+REDIS_CONFIG = Rails.application.config_for(:redis).freeze
+Redis.current = Redis.new(REDIS_CONFIG['redis'])
+
+require 'evss/claims_service'
+require 'facilities/async_client'
+require 'gi/client'
+require 'hca/configuration'
+require 'hca/service'
+require 'mvi/service'
+require 'rx/client'
+require 'sm/client'
+
+namespace :connectivity do
+  desc 'Check connectivity to all backend services'
+  task all: [:db, :edu, :evss, :gi, :hca, :logs, :mvi, :redis, :rx, :sm, :statsd, :vha]
+
+  desc 'Check DB'
+  task db: :environment do
+    begin
+      EVSSClaim::all
+      puts "DB connection success for #{Settings.database_url}."
+    rescue => e
+      puts "DB connection unsuccessful for #{Settings.database_url}!"
+      puts " - Error encountered: `#{e}`"
+    end
+  end
+
+  desc 'Check Edu SFTP'
+  task edu: :environment do
+    begin
+      Net::SFTP.start(Settings.edu.sftp.host, Settings.edu.sftp.user, password: Settings.edu.sftp.pass)
+      puts "Edu SFTP connection success for #{Settings.edu.sftp.host}."
+    rescue => e
+      puts "Edu SFTP connection unsuccessful for #{Settings.edu.sftp.host}!"
+      puts " - Error encountered: `#{e}`"
+    end
+  end
+
+  desc 'Check EVSS'
+  task evss: :environment do
+    begin
+      EVSS::ClaimsService.new({}).all_claims
+      # Should return an XML 403 response, which Faraday fails parsing,
+      # since it expects JSON
+      puts "EVSS connection super success for #{Settings.evss.url}!"
+    rescue Faraday::ParsingError
+      puts "EVSS connection success for #{Settings.evss.url}."
+    rescue => e
+      puts "EVSS connection unsuccessful for #{Settings.evss.url}!"
+      puts " - Error encountered: `#{e}`"
+    end
+  end
+
+  desc 'Check GI'
+  task gi: :environment do
+    begin
+      GI::Client.new.get_autocomplete_suggestions('university')
+      puts "GIDS connection success for #{Settings.gids.url}."
+    rescue => e
+      puts "GIDS connection unsuccessful for #{Settings.gids.url}!"
+      puts " - Error encountered: `#{e}`"
+    end
+  end
+
+  desc 'Check HCA'
+  task hca: :environment do
+    begin
+      HCA::Service.new.health_check
+      puts "HCA connection success for #{Settings.hca.endpoint}."
+    rescue => e
+      puts "HCA connection unsuccessful for #{Settings.hca.endpoint}!"
+      puts " - Error encountered: `#{e}`"
+    end
+  end
+
+  desc 'Check that logs are writeable'
+  task logs: :environment do
+    if File.writable?(Rails.root.join("log"))
+      puts "Logging directory is writeable."
+    else
+      puts "Logging directory is not writeable!"
+    end
+  end
+
+  desc 'Check MVI'
+  task mvi: :environment do
+    begin
+      user = User.new(
```

Comment:
Might you be able to re-use our `User` factory girl for this?  The down-side is that it's a little weird to create a dependency on `/spec/factories/` but the bonus is that if the `User` model ever changes, you won't need to update this .rake file.